src/components/grid/grid.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from '@emotion/styled';
+import customProperties from '../../../materials/custom-properties.json';
+import filterDataAttributes from '../../utils/filter-data-attributes';
+
+// TODO: check with design if those "breakpoints" are ok
+const minColumnSizes = {
+  1: '600px',
+  2: '450px',
+  3: '350px',
+  4: '275px',
+};
+
+const GridContainer = styled.div`
+  display: grid;
+  grid-gap: ${customProperties['--spacing-16']};
+  grid-auto-columns: 1fr;
+  grid-template-columns: ${props => {
+    switch (props.mode) {
+      case 'fixed':
+        return `repeat(${props.columns}, 1fr);`;
+      // By using `auto-fit` we ensure the columns have the same size and will wrap in case there is not enough space
+      // https://css-tricks.com/auto-sizing-columns-css-grid-auto-fill-vs-auto-fit/
+      case 'auto-fit': {
+        const minSize = minColumnSizes[props.columns];
+        if (!minSize) {
+          throw new Error(
+            `The number of columns ${props.columns} is not supported.`
+          );
+        }
+        return `repeat(auto-fit, minmax(${minSize}, 1fr));`;
+      }
+      default:
+        throw new Error(`Mode ${props.mode} is not supported.`);
+    }
+  }};
+`;
+
+const insertGridPlaceholders = (numberOfElements, limit) =>
+  numberOfElements > 0 &&
+  numberOfElements < limit &&
+  Array.from({
+    length: limit - numberOfElements,
+  }).map((_, index) => (
+    <div key={index} data-testid={`grid-placeholder-${index}`} />
+  ));
+
+const Grid = props => (
+  <GridContainer
+    mode={props.mode}
+    columns={props.columns}
+    {...filterDataAttributes(props)}
+  >
+    {props.children}
+    {insertGridPlaceholders(
+      React.Children.count(props.children),
+      props.columns
+    )}
+  </GridContainer>
+);
+Grid.displayName = 'Grid';
+Grid.propTypes = {
```

Comment:
I wonder about the API. The current one is pretty restrictive and it would suck if we had to upgrade the whole MC while we learn more about how to provide a flexible API. Exploring more possible use-cases now can probably avoid migration pain afterwards.

Maybe we can explore using Grid for more advanced use-cases before publishing it?

For example, we could try to build the [`CollapsiblePanel` header](https://user-images.githubusercontent.com/1765075/51261947-94ed5b80-19b1-11e9-8723-a6145dd47230.png) (if that's even something we intend it to be used for).

Reply:
Sure, we don't need to merge this in right away. We can keep it open for now.

> The current one is pretty restrictive

Definitely. There is a lot of room for improvements, this one is the bare minimum.